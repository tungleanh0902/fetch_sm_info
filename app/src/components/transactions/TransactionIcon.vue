<template>
  <div class="tx__icon">
    <img :src="icon" :alt="transactionType" />
  </div>
</template>

<script>
export default {
  name: `transaction-icon`,
  props: {
    transactionType: {
      type: String,
      required: true,
    },
  },
  computed: {
    icon() {
      return `/img/icons/activity/${this.transactionType}.svg`
    },
  },
}
</script>

<style>
.tx__icon {
  padding: 1rem;
}

.tx__icon img {
  max-height: 100%;
  max-width: 40px;
  display: block;
}

@media screen and (min-width: 1023px) {
  .tx__icon img {
    max-width: 48px;
  }
}
</style>
