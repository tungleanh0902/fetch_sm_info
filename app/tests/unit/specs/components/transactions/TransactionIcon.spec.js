import { shallowMount } from "@vue/test-utils"
import TransactionIcon from "src/components/transactions/TransactionIcon"

describe(`TransactionIcon`, () => {
  let wrapper

  const transactions = [
    {
      type: `Sent`,
    },
    {
      type: `Received`,
    },
    {
      type: `Staked`,
    },
    {
      type: `Restaked`,
    },
    {
      type: `Unstaked`,
    },
    {
      type: `Claimed`,
    },
    {
      type: `Voted`,
    },
    {
      type: `Deposit`,
    },
    {
      type: `Submitted`,
    },
  ]

  for (let i = 0; i < transactions.length; i++) {
    it(`renders a ${transactions[i].type} transaction icon`, () => {
      wrapper = shallowMount(TransactionIcon, {
        propsData: {
          transactionType: transactions[i].type,
        },
      })
      expect(wrapper.element).toMatchSnapshot()
      expect(wrapper.find(".tx__icon img").attributes("src")).toEqual(
        `/img/icons/activity/${transactions[i].type}.svg`
      )
    })
  }
})
