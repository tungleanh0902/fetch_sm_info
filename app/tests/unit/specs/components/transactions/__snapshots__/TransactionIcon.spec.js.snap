// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionIcon renders a Claimed transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Claimed"
    src="/img/icons/activity/Claimed.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Deposit transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Deposit"
    src="/img/icons/activity/Deposit.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Received transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Received"
    src="/img/icons/activity/Received.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Restaked transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Restaked"
    src="/img/icons/activity/Restaked.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Sent transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Sent"
    src="/img/icons/activity/Sent.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Staked transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Staked"
    src="/img/icons/activity/Staked.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Submitted transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Submitted"
    src="/img/icons/activity/Submitted.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Unstaked transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Unstaked"
    src="/img/icons/activity/Unstaked.svg"
  />
</div>
`;

exports[`TransactionIcon renders a Voted transaction icon 1`] = `
<div
  class="tx__icon"
>
  <img
    alt="Voted"
    src="/img/icons/activity/Voted.svg"
  />
</div>
`;
